packages/contracts/src/contracts/current/protocol/Exchange/MixinExchangeCore.sol

```diff
@@ -85,9 +83,15 @@ contract MixinExchangeCore is
         // Compute the order hash
         bytes32 orderHash = getOrderHash(order);
 
+        // Check if order has been cancelled
+        if (cancelled[orderHash]) {
+            LogError(uint8(Errors.ORDER_FULLY_FILLED_OR_CANCELLED), orderHash);
```

Comment:
Now that it's easier to differentiate between the fully filled and cancelled states, would it be worthwhile to separate `ORDER_FULLY_FILLED_OR_CANCELLED` into `ORDER_FULLY_FILLED` and `ORDER_CANCELLED`?